src/lib/PnP.Framework/Graph/SubscriptionsUtility.cs

```diff
@@ -27,24 +29,19 @@ public static Model.Subscription GetSubscription(string accessToken, Guid subscr
         {
             try
             {
-                // Use a synchronous model to invoke the asynchronous process
-                var result = Task.Run(async () =>
-                {
-                    var graphClient = GraphUtility.CreateGraphClient(accessToken, retryCount, delay, azureEnvironment: azureEnvironment);
+                var requestUrl = $"{GraphHttpClient.GetGraphEndPointUrl(azureEnvironment)}subscriptions/{subscriptionId}";
 
-                    var subscription = await graphClient.Subscriptions[subscriptionId.ToString()]
-                        .Request()
-                        .GetAsync();
+                var responseAsString = HttpHelper.MakeGetRequestForString(requestUrl, accessToken, retryCount: retryCount, delay: delay);
 
-                    var subscriptionModel = MapGraphEntityToModel(subscription);
-                    return subscriptionModel;
-                }).GetAwaiter().GetResult();
+                var response = JToken.Parse(responseAsString);
```

Comment:
Converting string to Model seems like it could be an extension method?

Reply:
Also tagging @KoenZomers as he wrote a lot of the original code